js/components/Star/index.js

```diff
@@ -0,0 +1,48 @@
+import React, { Component, PropTypes } from 'react';
+import {
+  TouchableOpacity,
+  Image,
+  StyleSheet,
+} from 'react-native';
+
+export default class Star extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      value: false,
+    };
+
+    this.toggle = this.toggle.bind(this);
+  }
+
+  toggle() {
+    this.setState({ value: !this.state.value });
+  }
+
+  render() {
+    const img = this.state.value ?
+      <Image source={require('./assets/star.png')}/> :
+      <Image source={require('./assets/star-border.png')}/>;
+
+    return (
+      <TouchableOpacity
+        style={styles.container}
+        onPress={this.toggle}>
```

Comment:
Yes that makes sense.
I would only argue that the scenes are container components and we don't need to wrap this Star component inside another container.